pkgs/build-support/node/fetch-npm-deps/default.nix

```diff
@@ -0,0 +1,52 @@
+{ lib, stdenv, rustPlatform, Security }:
+
+rec {
+  fetch-npm-deps = rustPlatform.buildRustPackage {
+    pname = "fetch-npm-deps";
+    version = (lib.importTOML ./Cargo.toml).package.version;
+
+    src = ./.;
+
+    cargoLock.lockFile = ./Cargo.lock;
+
+    buildInputs = lib.optional stdenv.isDarwin Security;
+  };
+
+  fetchNpmDeps =
+    { name ? "npm-deps"
+    , src ? null
+    , srcs ? [ ]
+    , patches ? [ ]
+    , sourceRoot ? [ ]
+    , hash ? ""
+    , ...
+    } @ args: stdenv.mkDerivation ({
+      nativeBuildInputs = [ fetch-npm-deps ];
+
+      buildPhase = ''
+        runHook preBuild
+
+        if [[ ! -f package-lock.json ]]; then
+          echo
+          echo "ERROR: The package-lock.json file does not exist!"
+          echo
+          echo "package-lock.json is required to make sure that npmDepsHash doesn't change"
+          echo "when packages are updated on npm."
+          echo
+
+          exit 1
+        fi
+
+        fetch-npm-deps package-lock.json $out
+
+        cp package-lock.json $out
+
+        runHook postBuild
+      '';
+
+      dontInstall = true;
```

Comment:
Should probably split up the phases a bit more and copy the relevant files to `$out` in `installPhase`